blockchain/blockchain.go

```diff
@@ -57,31 +57,19 @@ func NewProcessor(boltDB *bolt.DB) (Processor, error) {
 	}
 
 	// Setup ethereum client
-	if client, err := rpc.Dial(config.GetString(config.EthereumJsonRpcEndpointKey)); err != nil {
+
+	if ethclients, err := GetEthereumClient(); err != nil {
 		return p, errors.Wrap(err, "error creating RPC client")
 	} else {
-		p.rawClient = client
-		p.ethClient = ethclient.NewClient(client)
+		p.rawClient = ethclients.RawClient
+		p.ethClient = ethclients.EthClient
 	}
-
+	defer ethereumClient.Close()
```

Comment:
```suggestion
```

Reply:
We treat EthereumClient also as a Component ? 
It should be used by serviceMetaData and Processor , EthereumClient should also be closed in the Close() method of Components , this way we handle the scenarios  gracefully
so 
1) add Method EthereumClient() as a Component
2) close() from serviceMetaData and Processor 
3) Gracefully call the close on Component
4) component.EthereumClient() will be passed on to ServiceMetaData and Processor

Reply:
Great, thanks, I like this idea very much!